pkg/sql/gcjob/gc_job.go

```diff
@@ -84,6 +90,150 @@ func performGC(
 	return nil
 }
 
+// unsplitRange unsets the sticky bit of a range if it's manually split
+func unsplitRange(
+	ctx context.Context, execCfg *sql.ExecutorConfig, desc roachpb.RangeDescriptor,
+) error {
+	if !execCfg.Codec.ForSystemTenant() {
+		return nil
+	}
+
+	if !desc.GetStickyBit().IsEmpty() {
+		// Swallow "key is not the start of a range" errors because it would mean
+		// that the sticky bit was removed and merged concurrently. DROP TABLE
+		// should not fail because of this.
+		if err := execCfg.DB.AdminUnsplit(ctx, desc.StartKey); err != nil &&
+			!strings.Contains(err.Error(), "is not the start of a range") {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// unsplitRangesForTable unsplit any manually split ranges within the table span.
+func unsplitRangesForTable(
+	ctx context.Context, execCfg *sql.ExecutorConfig, tableDesc catalog.TableDescriptor,
+) error {
+	// Gate this on being the system tenant because secondary tenants aren't
+	// allowed to scan the meta ranges directly.
+	if !execCfg.Codec.ForSystemTenant() {
+		return nil
+	}
+
+	span := tableDesc.TableSpan(execCfg.Codec)
+	ranges, err := kvclient.ScanMetaKVs(ctx, execCfg.DB.NewTxn(ctx, "unsplit-ranges-for-table"), span)
+	if err != nil {
+		return err
+	}
+	for _, r := range ranges {
+		var desc roachpb.RangeDescriptor
+		if err := r.ValueProto(&desc); err != nil {
+			return err
+		}
+
+		if err := unsplitRange(ctx, execCfg, desc); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
```

Comment:
this is almost identical to the function in the `sql` package hanging off of the `planner`. Can you refactor that thing to be a free-standing function which is exported and then invoke it here?